balancer/rls/picker_test.go

```diff
@@ -699,13 +723,24 @@ func (s) TestPick_DataCacheHit_PendingEntryExists_ExpiredEntry(t *testing.T) {
 
 			// Make sure an RLS request is sent out.
 			verifyRLSRequest(t, rlsReqCh, true)
-
-			// At this point, we have a cache entry with a small maxAge, and the RLS
-			// server is configured to block on further RLS requests. As we retry the
-			// RPC, at some point the cache entry would expire and force us to send an
-			// RLS request. But this request would exceed the deadline since the
-			// server blocks.
-			makeTestRPCAndVerifyError(ctx, t, cc, codes.DeadlineExceeded, context.DeadlineExceeded)
+			firstRPCDone.Fire()
+
+			// At this point, we have a cache entry with a small maxAge, and the
+			// RLS server is configured to block on further RLS requests. As we
+			// retry the RPC, at some point the cache entry would expire and
+			// force us to send an RLS request which would block on the server,
+			// giving us a pending cache entry for the duration of the test.
+			go func() {
+				client := testgrpc.NewTestServiceClient(cc)
+				for {
+					select {
+					case <-ctx.Done():
+						return
+					default:
+						client.EmptyCall(ctx, &testpb.Empty{})
+					}
+				}
```

Comment:
I believe this is equivalent?

```go
for ctx.Err() == nil {
	client.EmptyCall(ctx, &testpb.Empty{})
}
```

Should there be a small delay between iterations to avoid a busy loop?

Reply:
Done.